src/all/nijiero/src/eu/kanade/tachiyomi/extension/all/nijiero/Nijiero.kt

```diff
@@ -0,0 +1,148 @@
+package eu.kanade.tachiyomi.extension.all.nijiero
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Nijiero : ParsedHttpSource() {
+    override val name = "Nijiero"
+    override val lang = "all"
+    override val supportsLatest = true
+
+    override val baseUrl = "https://www.nijiero-ch.com"
+
+    override val client = network.client.newBuilder().followRedirects(true).build()
+	
+    class TagFilter : Filter.Select<String>("Category", nijieroTags)
+
+    override fun pageListParse(document: Document): List<Page> {
+        return document.select("#entry > ul > li a[href]").mapIndexed { i, linkElement ->
+            val linkUrl = linkElement.attr("href").removeSuffix(".webp").removePrefix(baseUrl)
+            Page(i, document.location(), linkUrl)
+        }
+    }
+
+    override fun imageUrlParse(document: Document): String = throw UnsupportedOperationException()
+	
+    private val spaceRegex = Regex("""\s+""")
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val tagIdx: Int = (filters.last() as TagFilter).state
+
+        val keyword = when {
+            query.isBlank() -> nijieroTags[tagIdx]
+            else -> query
+        }.replace(spaceRegex, "-").lowercase()
+
+        val uniqueParam = System.currentTimeMillis()
+        var url = baseUrl.toHttpUrl().newBuilder()
+            .addEncodedPathSegment("category")
+            .addEncodedPathSegment(keyword)
+            .addEncodedPathSegment("page")
+            .addEncodedPathSegment(page.toString())
+            .addQueryParameter("refresh", uniqueParam.toString())
+            .build()
+		
+        // if 404, recheck
+        if (client.newCall(GET(url, headers)).execute().code == 404) {
```

Comment:
it would be better to do this in an interceptor, so you can simply return the response as is if available or make new request and return it's response

Reply:
This code should actually check if the current page was not found as a tag, so it tries again with category... not sure what you mean by that, sorry.

Reply:
I mean instead of make a call just to check if it exists, add an [interceptor](https://square.github.io/okhttp/features/interceptors/) to detect it and make new call there.

basically return the tag request from here, in the interceptor prceed the call and check if it is 404 and url contains tag (this request) and if so, modify the request and call again, otherwise just return the original response